server.go

```diff
@@ -462,11 +462,85 @@ func (p *sshFxpOpenPacket) respond(svr *Server) responsePacket {
 		osFlags |= os.O_EXCL
 	}
 
-	f, err := os.OpenFile(svr.toLocalPath(p.Path), osFlags, 0o644)
+	name := svr.toLocalPath(p.Path)
+	mode := os.FileMode(0o644)
+
+	var applyAttrs []func(*os.File) error
+
+	// Both `sshFileXferAttrPermissions` and `sshFileXferAttrACmodTime` are set
+	// by e.g. `sftp`. Just in case, we handle all other cases as well.
+	// Only apply for newly created files.
+	var err error
+	useAttrs := true
+	if _, err := os.Stat(name); err == nil {
+		useAttrs = false
+	}
+	if b, ok := p.Attrs.([]byte); useAttrs && ok {
+		if (p.Flags & sshFileXferAttrSize) != 0 {
```

Comment:
We’re just going to entirely repeat `unmarshalFileStat`? This probably isn’t a good idea.

Reply:
I saw this method, but it wasn't being used by any of the `Server` implementations, so I omitted it. I can switch to using that method if you prefer.